### PR TITLE
Feature/617 search resources on map

### LINKED ISSUE #617: Show search page resources on a map
Tasks in this issue:
- [x] Display a map
- [x] Show search results as markers on the map
- [x] Add a toggle to switch between the map view and search results

The location can be found from unit data. The resources have a location field as well, but it seems to be null most of the time.

This issue is only for making the "mvp" version of the map.
Styling and more advanced functionality will be done in later issues. 
Now the map can be behind a toggle. Later it will be shown at the same time with the results listing so this should be kept in mind when implementing the Map container.
Below are the current designs on how the search results would look in the end with the map and new Helsinki brand:

Map closed:

varaamo2-lista

Map open:

varaamo2-kartta

### PR DESCRIPTION
**Pull request by YoYuUm**
Closes #617.

### CODE REVIEW

File: app/pages/search/results/SearchResults.spec.js [hunk 1]
```diff
@@ -12,6 +13,8 @@ describe('pages/search/results/SearchResults', () => {
   const defaultProps = {
     isFetching: false,
     searchResultIds: Immutable(['resource-1', 'resource-2']),
+    onToggleMap: () => {},
```
Comment (lraivio) [inline app/pages/search/results/SearchResults.spec.js]: - [ ] alphabetic order.

File: app/pages/search/searchPageSelector.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ import requestIsActiveSelectorFactory from 'state/selectors/factories/requestIsA
 
 const searchDoneSelector = state => state.ui.search.searchDone;
 const searchResultIdsSelector = state => state.ui.search.results;
+const searchShowMapSelector = state => state.ui.search.showMap;
```
Comment (lraivio) [inline app/pages/search/searchPageSelector.js]: - [ ] This can be named just `showMapSelector`.

File: app/assets/styles/app.less [hunk 3]
```diff
@@ -43,3 +43,7 @@
 
 // Customizations
 @import './customization/espoo/customization';
+
+// Leaflet
+// React-Select
```
Comment (lraivio) [inline app/assets/styles/app.less]: - [ ] Remove this comment.

File: app/assets/styles/app.less [hunk 4]
```diff
@@ -43,3 +43,7 @@
...
+@import '~leaflet/dist/leaflet.css';
```
Comment (lraivio) [inline app/assets/styles/app.less]: - [ ] This should be moved higher in the file where the other third party css imports are located.

File: app/shared/resource-map/map.less [hunk 5]
```diff
@@ -0,0 +1,5 @@
+@import "./popup/popup.less";
```
Comment (lraivio) [inline app/shared/resource-map/map.less]: - [ ] Remove this import.

File: app/shared/shared.less [hunk 6]
```diff
@@ -8,6 +8,7 @@
 @import './footer/footer';
 @import './form-fields/form-fields';
 @import './info-popover/info-popover';
+@import './map/map';
```
Comment (lraivio) [inline app/shared/shared.less]: - [ ] The path is wrong. Should be './resource-map/map'

File: app/shared/resource-map/MapContainer.spec.js [hunk 7]
```diff
@@ -22,21 +21,16 @@ describe('screens/map/MapContainer', () => {
...
-
   it('renders a Leaflet Map', () => {
     const map = getWrapper().find(Map);
     expect(map).to.have.length(1);
   });
 
-  it('renders Markers', () => {
-    const markers = [{ id: '1', longitude: 1, latitude: 1, owner: 'A' }];
-    const element = getWrapper({ markers }).find(Markers);
+  it('renders Marker', () => {
```
Comment (lraivio) [inline app/shared/resource-map/MapContainer.spec.js]: - [ ] You could test that a marker is rendered for each marker in props.

File: app/shared/resource-map/Marker.spec.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import React from 'react';
+import { Marker } from 'react-leaflet';
+import { browserHistory } from 'react-router';
+import simple from 'simple-mock';
+
+import Markers from './Marker';
```
Comment (lraivio) [inline app/shared/resource-map/Marker.spec.js]: - [ ] Change `Markers` to `Marker`.

**Comment by YoYuUm** [comment]
Now it's squashed :)

**Pull request merged**